### PR TITLE
Generate C preprocessor code to hide things not on Windows

### LINKED ISSUE #804: C API doesn't mention platform restrictions regarding trampoline
### Describe the bug
The wasmer.h and wasmer.hh headers list the structs and functions associated with the trampoline features, however this feature is only enabled on unix x64 platforms.
The tests runtime-c-api/test-import-function and runtime-c-api/test-import-function-callinfo do not link on windows.

### Expected behavior
The header files should state that these aren't available or even better, remove them with macros.

### PR DESCRIPTION
**Pull request by MarkMcCaskey**
resolves #804 

# Review

- [x] Add a short description of the the change to the CHANGELOG.md file

### CODE REVIEW

File: lib/runtime-c-api/wasmer.hh [hunk 1]
```diff
@@ -612,32 +635,46 @@ uint32_t wasmer_table_length(wasmer_table_t *table);
 /// and `wasmer_last_error_message` to get an error message.
 wasmer_result_t wasmer_table_new(wasmer_table_t **table, wasmer_limits_t limits);
 
+#if (!defined(_WIN32) && defined(ARCH_X86_64))
```
Comment (nlewycky) [inline lib/runtime-c-api/wasmer.hh]: Optional: these are clearly a group, just put a single #if around them to disable all these related trampoline functions for windows?

Comment (MarkMcCaskey) [reply lib/runtime-c-api/wasmer.hh]: Yeah, it's unfortunate that it did that; if we manually edit the header file though then we have to do it after every time it's generated

Comment (MarkMcCaskey) [reply lib/runtime-c-api/wasmer.hh]: I put it at the top level in Rust

**Pull request merged**